Realm/ObjectServerTests/SwiftAsymmetricSyncServerTests.swift

```diff
@@ -0,0 +1,249 @@
+    @Persisted public var decimalMap: Map<String, Decimal128>
+    @Persisted public var uuidMap: Map<String, UUID>
+    @Persisted public var anyMap: Map<String, AnyRealmValue>
+
+    public override class func _realmIgnoreClass() -> Bool {
+        return true
+    }
+
+    public convenience init(id: ObjectId, string: String, int: Int, bool: Bool) {
+        self.init()
+        self._id = id
+        self.string = string
+        self.int = int
+        self.bool = bool
+    }
+}
+
+class SwiftAsymmetricSyncTests: SwiftSyncTestCase {
+    override class var defaultTestSuite: XCTestSuite {
+        // async/await is currently incompatible with thread sanitizer and will
+        // produce many false positives
+        // https://bugs.swift.org/browse/SR-15444
+        if RLMThreadSanitizerEnabled() {
+            return XCTestSuite(name: "\(type(of: self))")
+        }
+        return super.defaultTestSuite
+    }
+
+    var asymmetricApp: App {
+        var appId = SwiftAsymmetricSyncTests.asymmetricAppId
+        if appId == nil {
+            do {
+                let objectSchema = RLMObjectSchema(forObjectClass: SwiftObjectAsymmetric.self)
+                appId = try RealmServer.shared.createAppForAsymmetricSchema([objectSchema])
+                SwiftAsymmetricSyncTests.asymmetricAppId = appId
+            } catch {
+                XCTFail("Failed to create Asymmetric app: \(error)")
+            }
+        }
+
+        let appConfig = AppConfiguration(baseURL: "http://localhost:9090",
+                                         transport: nil,
+                                         localAppName: nil,
+                                         localAppVersion: nil)
+        return App(id: appId!, configuration: appConfig)
+    }
+    static var asymmetricAppId: String?
+
+    override func tearDown() {
+        let user = try! logInUser(for: .anonymous, app: asymmetricApp)
+        let mongoClient = user.mongoClient("mongodb1")
+        let database = mongoClient.database(named: "test_data")
+        let collection =  database.collection(withName: "SwiftObjectAsymmetric")
+        removeAllFromCollection(collection)
+        super.tearDown()
+    }
+
+    func testAsymmetricObjectSchema() throws {
+        var configuration = (try logInUser(for: basicCredentials(app: asymmetricApp), app: asymmetricApp)).flexibleSyncConfiguration()
+        configuration.objectTypes = [SwiftObjectAsymmetric.self]
+        let realm = try Realm(configuration: configuration)
+        XCTAssertTrue(realm.schema.objectSchema[0].isAsymmetric)
+    }
+
+    func testOpenLocalRealmWithAsymmetricObjectError() throws {
+        let configuration = Realm.Configuration(objectTypes: [SwiftObjectAsymmetric.self])
+        do {
+            _ = try Realm(configuration: configuration)
+            XCTFail("Opening a local Realm with an `Asymmetric` table should fail")
+        } catch {
+            XCTAssertNotNil(error)
```

Comment:
As with the obj-c versions these should be testing that the error is the correct error.